lib/models/contact.js

```diff
@@ -142,9 +142,20 @@ ContactSchema.methods.recordPoints = function(points) {
   return this;
 };
 
+/**
+ * Will reset the timeout rate after 72 hours without a timeout.
+ */
+ContactSchema.methods.resetTimeoutRate = function() {
+  const now = Date.now();
+  const window = 259200000; // 72 hours
+  if (this.lastTimeout < now - window) {
```

Comment:
Date.now() is expensive. How about this:
`if (this.timeoutRate && this.lastTimeout < Date.now() - window)`

That should call Date.now() only if the farmer has a timeoutRate that could be reset.